compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -479,10 +479,13 @@ impl<'a> Resolver<'a> {
                     ),
                 );
                 err.help(
-                    &format!("it is currently only allowed to use either `{0}` or `{{ {0} }}` as generic constants", name)
+                    &format!("note: type parameters are currently not permitted in anonymous constants")
                 );
                 err
             }
+            ResolutionError::ParamInNonTrivialAnonConst(None) => {
+                todo!()
```

Comment:
I'm unsure what to do here, or where it'll be needed.

Reply:
I saw you only construct this with `Some` - why make it an `Option` then?

Reply:
We changed it to include a `bool`, which is now being used.